src/apps/dolomite/arbitrum/dolomite.pools.token-fetcher.ts

```diff
@@ -0,0 +1,93 @@
+import { Inject } from '@nestjs/common';
+import { BigNumberish } from 'ethers';
+
+import { IAppToolkit, APP_TOOLKIT } from '~app-toolkit/app-toolkit.interface';
+import { PositionTemplate } from '~app-toolkit/decorators/position-template.decorator';
+import {
+  buildDollarDisplayItem,
+  buildPercentageDisplayItem,
+} from '~app-toolkit/helpers/presentation/display-item.present';
+import { DOLOMITE_AMM_FACTORY_ADDRESSES } from '~apps/dolomite/utils';
+import {
+  UniswapV2PoolOnChainTemplateTokenFetcher,
+  UniswapV2TokenDataProps,
+} from '~apps/uniswap-v2/common/uniswap-v2.pool.on-chain.template.token-fetcher.ts';
+import { GetDisplayPropsParams } from '~position/template/app-token.template.types';
+
+import { DolomiteAmmFactory, DolomiteAmmPair, DolomiteContractFactory } from '../contracts';
+
+@PositionTemplate()
+export class ArbitrumDolomitePoolsTokenFetcher extends UniswapV2PoolOnChainTemplateTokenFetcher<
```

Comment:
You could have extend this one instead `UniswapV2DefaultPoolOnChainTemplateTokenFetcher`
```
@PositionTemplate()
export class ArbitrumDolomitePoolTokenFetcher extends UniswapV2DefaultPoolOnChainTemplateTokenFetcher {
  groupLabel = 'Pools';
  factoryAddress = '__FACTORY_ADDRESS__';
}
```

Reply:
There were sufficient-enough changes that needed to be made because we renamed our smart contract variables that I felt it was fine to not extend from the default class